File: app/src/main/java/com/shevart/androidcorelearn/multi_threading/executors/SimpleQueueExecutorDemo.java
package com.shevart.androidcorelearn.multi_threading.executors;

import com.shevart.androidcorelearn.common.DemoStartable;
import com.shevart.androidcorelearn.utils.ThreadDemoUtil;

class SimpleQueueExecutorDemo implements DemoStartable {
    private SimpleAsyncQueueExecutor simpleAsyncQueueExecutor;

    @Override
    public void startDemo() {
        if (simpleAsyncQueueExecutor == null) {
            simpleAsyncQueueExecutor = new SimpleAsyncQueueExecutor();
        }

        ThreadDemoUtil.sendEmptyTasksToExecutor5(simpleAsyncQueueExecutor);
    }
}

File: app/src/main/java/com/shevart/androidcorelearn/multi_threading/executors/SimpleQueueExecutorDemo.kt
package com.shevart.androidcorelearn.multi_threading.executors

import com.shevart.androidcorelearn.common.DemoStartable
import com.shevart.androidcorelearn.utils.ThreadDemoUtil

internal class SimpleQueueExecutorDemo : DemoStartable {
    private var simpleAsyncQueueExecutor: SimpleAsyncQueueExecutor? = null

    override fun startDemo() {
        if (simpleAsyncQueueExecutor == null) {
            simpleAsyncQueueExecutor = SimpleAsyncQueueExecutor()
        }

        ThreadDemoUtil.sendEmptyTasksToExecutor5(simpleAsyncQueueExecutor!!)
    }
}